pandas/core/util/hashing.py

```diff
@@ -276,3 +310,63 @@ def hash_array(vals, encoding='utf8', hash_key=None, categorize=True):
     vals *= np.uint64(0x94d049bb133111eb)
     vals ^= vals >> 31
     return vals
+
+
+def _hash_scalar(val, encoding='utf8', hash_key=None):
+    """
+    Hash scalar value
+
+    Returns
+    -------
+    1d uint64 numpy array of hash value, of length 1
+    """
+
+    if hash_key is None:
+        hash_key = _default_hash_key
+
+    if isnull(val):
+        # this is to be consistent with the _hash_categorical implementation
+        return np.array([np.iinfo(np.uint64).max], dtype='u8')
+
+    if isinstance(val, string_and_binary_types + (text_type,)):
+        vals = np.array([val], dtype=object)
+        string_like = True
+    else:
+        vals = np.array([val])
+        string_like = False
+
+    dtype = vals.dtype
+
+    #dtype, vals = infer_dtype_from_array([vals])
+    #if dtype == np.object_:
+    #    vals = np.asarray(vals, dtype='object')
+    #    dtype = vals.dtype
+
+    # we'll be working with everything as 64-bit values, so handle this
+    # 128-bit value early
+    if np.issubdtype(dtype, np.complex128):
+        return hash_array(vals.real) + 23 * hash_array(vals.imag)
+
+    # First, turn whatever array this is into unsigned 64-bit ints, if we can
+    # manage it.
+    elif isinstance(dtype, np.bool):
+        vals = vals.astype('u8')
+    elif issubclass(dtype.type, (np.datetime64, np.timedelta64)):
+        vals = vals.view('i8').astype('u8', copy=False)
+    elif issubclass(dtype.type, np.number) and dtype.itemsize <= 8:
+        vals = vals.view('u{}'.format(vals.dtype.itemsize)).astype('u8')
+    else:
+        if not string_like:
+            from pandas import Index
+            vals = Index(vals).values
```

Comment:
This is also a bit an ugly part. I use `Index(vals)` to get the correct type coercion (eg for a `Timestamp` object, to ensure it does the same as `hash_array`). But I don't do this in the beginning because this is much slower than needed for simple numerical values. 
It would be nice to have a utility function to convert a list of values to an array with the dtype "how pandas wants it" (but maybe that exists and I just don't know about it)